crates/agglayer-contracts/src/lib.rs

```diff
@@ -1,14 +1,16 @@
 //! Agglayer smart-contract bindings.
 
-use std::sync::Arc;
+use std::{collections::HashMap, sync::Arc};
 
 use agglayer_primitives::U256;
 use alloy::{
     eips::BlockNumberOrTag,
     primitives::{Address, FixedBytes, TxHash, B256},
     providers::Provider,
     rpc::types::{Filter, TransactionReceipt},
+    signers::k256::elliptic_curve::ff::derive::bitvec::macros::internal::funty::Fundamental,
 };
+use tokio::sync::RwLock;
```

Comment:
Would this work with a std `RwLock`? As far as I can tell, it is not locked across an `await`, and there all sorts of extra complexities (and also perf overhead) associated with async locks.

Reply:
> Maybe. But with async RW lock we are on the safe side.

Not quite. Using `std::sync::RwLock` across await is a compiler error so there is no safety concern just changing the import. Worst case it fails to compile and switch back. IIRC, official [tokio docs suggest](https://tokio.rs/tokio/tutorial/shared-state#on-using-stdsyncmutex-and-tokiosyncmutex) using std locks and only switching to tokio locks if needed.